benchmark/single-source/ProtocolConformance.swift

```diff
@@ -0,0 +1,59 @@
+//===--- ProtocolDispatch.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let ProtocolConformance = BenchmarkInfo (
+  name: "ProtocolConformance",
+  runFunction: run_ProtocolConformance,
+  tags: [.validation, .runtime])
+
+protocol P {}
+
+struct One: P {}
+struct Two {}
+
+struct Cat<T, U> {}
+
+extension Cat: P where T: P, U: P {}
+
+protocol Growable {}
+extension Growable {
+  func grow() -> (Growable, Growable) {
+    return (Cat<Self, One>(), Cat<Self, Two>())
+  }
+}
+
+extension One: Growable {}
+extension Two: Growable {}
+extension Cat: Growable {}
+
+@inline(never)
+public func run_ProtocolConformance(_ N: Int) {
+  var array: [Growable] = [One(), Two()]
+  var count = 0
+  var i = 0
+  var checks = 0
+  while checks < N * 1_000 {
+    let (a, b) = array[i].grow()
+    array.append(a)
+    array.append(b)
+    for elt in array {
+      if elt is P {
+        count += 1
+      }
+      checks += 1
+    }
+    i += 1
+  }
+  blackHole(count)
```

Comment:
I know this isn't a correctness test, but would it be difficult to do this instead of `blackHole`?
`CheckResults(some_formula(checks) == count)

Reply:
Great idea. Took me a minute to figure out how to calculate the expected number, but I got it.